UPGRADE_GUIDE.md

```diff
@@ -2,6 +2,10 @@
 
 ## From `0.3.5` to `0.3.6`
 
+### Multithreaded connection removed
+
+The old `MultiThreadedSpatialOSConnectionHandler` has been removed and the method `SpatialOSConnectionHandlerBuilder.SetThreadingMode` has been removed to reflect this change.
```

Comment:
Not sure if this is needed in the upgrade guide? Perhaps mention in the changelog itself that the `SetThreadingMode` method has been removed